internal/blocktx/store/postgresql/postgres.go

```diff
@@ -66,14 +82,65 @@ func New(dbInfo string, idleConns int, maxOpenConns int, opts ...func(postgreSQL
 }
 
 func (p *PostgreSQL) Close() error {
-	return p.db.Close()
+	return p._db.Close()
 }
 
 func (p *PostgreSQL) Ping(ctx context.Context) error {
-	r, err := p.db.QueryContext(ctx, "SELECT 1;")
+	r, err := p._db.QueryContext(ctx, "SELECT 1;")
 	if err != nil {
 		return err
 	}
 
 	return r.Close()
 }
+
+func (p *PostgreSQL) StartUnitOfWork(ctx context.Context) (store.UnitOfWork, error) {
+	tx, err := p._db.BeginTx(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// This will create a clone of the store and use the transaction created
+	// above to avoid messing with the state of the main singleton store
+	cloneStore := &PostgreSQL{
+		_tx:                       tx,
+		db:                        tx,
+		now:                       time.Now,
+		maxPostgresBulkInsertRows: maxPostgresBulkInsertRows,
+		tracingEnabled:            p.tracingEnabled,
+		tracingAttributes:         p.tracingAttributes,
+	}
+
+	return cloneStore, nil
+}
+
+// UnitOfWork methods below
+func (p *PostgreSQL) Commit() error {
```

Comment:
Add one new line between function and comment as the comment does not describe the function

Reply:
Unit of Work was removed